adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceRtbRewardedAd.m

```diff
@@ -0,0 +1,166 @@
+         @"Using the default instance ID."];
+        self.instanceID = GADMIronSourceDefaultRtbInstanceId;
+    }
+    
+    NSString *bidResponse = adConfiguration.bidResponse;
+    NSMutableDictionary<NSString *, NSString *> *extraParams = [GADMAdapterIronSourceUtils getExtraParamsWithWatermark:adConfiguration.watermark];
+    
+    ISARewardedAdRequest *adRequest = [[[[ISARewardedAdRequestBuilder alloc] initWithInstanceId: self.instanceID adm: bidResponse] withExtraParams:extraParams] build];
+    
+    [ISARewardedAdLoader loadAdWithAdRequest: adRequest delegate: self];
+}
+
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    [GADMAdapterIronSourceUtils onLog:NSStringFromSelector(_cmd)];
+    id<GADMediationRewardedAdEventDelegate> rewardedDelegate = self.rewardedAdEventDelegate;
+
+    if (!self.biddingISARewardedAd){
+        if (rewardedDelegate){
+            NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                              GADMAdapterIronSourceErrorFailedToShow,
+                                                                              @"the ad is nil");
+            [rewardedDelegate didFailToPresentWithError:error];
+        }
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString stringWithFormat:@"Failed to show due to ad not loaded, for Instance ID: %@",
+                self.instanceID]];
+        return;
+    }
+    
+    [self.biddingISARewardedAd setDelegate:self];
+    [self.biddingISARewardedAd showFromViewController:viewController ];
+}
+
+- (void)rewardedAdDidLoad:(nonnull ISARewardedAd *)rewardedAd {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ instanceId= %@ adId= %@", NSStringFromSelector(_cmd),
+            rewardedAd.adInfo.instanceId,
+            rewardedAd.adInfo.adId]];
+    self.biddingISARewardedAd = rewardedAd;
+    if (!self.rewardedAdLoadCompletionHandler){
+        return;
+    }
+    
+    self.rewardedAdEventDelegate = self.rewardedAdLoadCompletionHandler(self, nil);
+}
+
+- (void)rewardedAdDidFailToLoadWithError:(nonnull NSError *)error {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ with error= %@", NSStringFromSelector(_cmd), error.localizedDescription]];
+    if (!self.rewardedAdLoadCompletionHandler){
+        return;
+    }
+    self.rewardedAdLoadCompletionHandler(nil, error);
+}
+
+- (void)rewardedAdDidShow:(nonnull ISARewardedAd *)rewardedAd {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"%@ instanceId= %@ adId= %@", NSStringFromSelector(_cmd),
+            rewardedAd.adInfo.instanceId,
+            rewardedAd.adInfo.adId]];
+    id<GADMediationRewardedAdEventDelegate> rewardedDelegate = self.rewardedAdEventDelegate;
+    if (!rewardedDelegate){
+        return;
+    }
+    
+    [rewardedDelegate willPresentFullScreenView];
```

Comment:
I suggest moving this to the end of presentFromViewController (as explained in the comment on GADMAdapterIronSourceRtbInterstitialAd.m)

Reply:
Done